components/terraform/vault-deployment/helm-install.tf

```diff
@@ -1,10 +1,8 @@
 
 resource "kubernetes_secret" "vault_licence" {
-  depends_on = [kubernetes_namespace.playground-namespace]
-
   metadata {
     name      = "hashicorp-vault-license"
-    namespace = var.namespace
+    namespace = kubernetes_namespace.vault.metadata[0].name
```

Comment:
- the Vault license needs to be deployed to the dedicated vault namespace
- here we use a direct reference via name of the namespace resource to resolve the dependency
- I'm not in favor of this method here, compared to depends_on in the secret above, because it's rather ugly with the key [0] selection, but meh